src/Executor/Executor.php

```diff
@@ -327,8 +331,11 @@ public function call(string $method, string $path = '', array $headers = [], arr
             }
         }
 
-        if ((curl_errno($ch)/* || 200 != $responseStatus*/)) {
-            throw new Exception(curl_error($ch) . ' with status code ' . $responseStatus, $responseStatus);
+        if ($curlError) {
+            if ($curlError == CURLE_OPERATION_TIMEDOUT) {
+                throw new AppwriteException(AppwriteException::FUNCTION_SYNCHRONOUS_TIMEOUT, $curlErrorMessage, Response::STATUS_CODE_REQUEST_TIMEOUT);
```

Comment:
once you add this error config to errors.php, you dont need to pass in the status code here. Its picked up automatically from the config

Reply:
Also what is the need to handle this case specifically?

Reply:
Added. We handle the case specifically to throw a specific exception, in order to display a error page to the user, instead of a 500 with no body.

Reply:
okay cool 👍